mod/wikirate_source/set/type/source/source.rb

```diff
@@ -175,66 +178,68 @@ def parse_source_page url
 end
 
 format :html do
-
   def edit_slot args
     # see claim.rb for explanation of core_edit
-    super args.merge(:core_edit=>true)
+    super args.merge(core_edit: true)
   end
 
   view :metric_import_link do |args|
-    file_card = Card[card.name+"+File"]
-    if file_card and mime_type = file_card.file.content_type and ( mime_type == "text/csv" || mime_type == "text/comma-separated-values" )
-      card_link file_card, {:text=>"Import to metric values",:path_opts=>{:view=>:import}}
-    else
-      ""
-    end
+    file_card = Card[card.name+'+File']
+    return '' unless has_import_mime_type?(file_card)
+    card_link file_card, text: "Import to metric values",
+                         path_opts: { view: :import }
+  end
+
+  def has_import_mime_type? file_card
+    file_card && (mime_type = file_card.file.content_type) &&
+      (mime_type == 'text/csv' || mime_type == 'text/comma-separated-values')
   end
 
   view :original_link do |args|
     with_original do |card, type|
       case type
       when :file
-        link_to (args[:title] || "Download"),card.file.url
+        link_to (args[:title] || 'Download'),card.file.url
       when :link
-        link_to (args[:title] || "Visit Source"),card.content
+        link_to (args[:title] || 'Visit Source'),card.content
       when :text
-        card_link card, :text=>(args[:title] || "Visit Text Source")
+        card_link card, text: (args[:title] || 'Visit Text Source')
       end
     end
   end
 
   view :original_icon_link do |args|
-    _render_original_link args.merge(:title=>content_tag(:i, '', :class=>"fa fa-#{icon}"))
+    title = content_tag(:i, '', class: "fa fa-#{icon}")
+    _render_original_link args.merge(title: title)
   end
 
   def with_original
-    if file_card = card.fetch(:trait=>:file )
+    if file_card = card.fetch(trait: :file )
```

Comment:
Assignment in condition - you probably meant to use `==`.<br>Space inside parentheses detected.